Assets/MixedRealityToolkit/_Core/Definitions/InputSystem/MixedRealityInputAction.cs

```diff
@@ -51,5 +52,48 @@ public MixedRealityInputAction(uint id, string description, AxisType axisConstra
 
         [SerializeField]
         private AxisType axisConstraint;
+
+        public static bool operator ==(MixedRealityInputAction left, MixedRealityInputAction right)
+        {
+            return left.Equals(right);
+        }
+
+        public static bool operator !=(MixedRealityInputAction left, MixedRealityInputAction right)
+        {
+            return !left.Equals(right);
+        }
+
+        #region IEqualityComparer Implementation
+
+        bool IEqualityComparer.Equals(object left, object right)
+        {
+            if (ReferenceEquals(null, left) || ReferenceEquals(null, right)) { return false; }
+            if (!(left is MixedRealityInputAction) || !(right is MixedRealityInputAction)) { return false; }
+            return ((MixedRealityInputAction)left).Equals((MixedRealityInputAction)right);
+        }
+
+        public bool Equals(MixedRealityInputAction other)
+        {
+            return Id == other.Id &&
+                   AxisConstraint == other.AxisConstraint;
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (ReferenceEquals(null, obj)) { return false; }
+            return obj is MixedRealityInputAction && Equals((MixedRealityInputAction)obj);
+        }
+
+        int IEqualityComparer.GetHashCode(object obj)
+        {
+            return obj is MixedRealityInputAction ? ((MixedRealityInputAction)obj).GetHashCode() : 0;
+        }
+
+        public override int GetHashCode()
+        {
+            return base.GetHashCode();
```

Comment:
Let's do a custom GetHashCode implementation...

Ex:
return string.Format("{0}.{1}", Id, AxisConstraint).GetHashCode();

Reply:
Is there a particular reason for that @davidkline-ms ?

in any case, we have may implementations with hashes so far, so if this is required, there should be a separate task to discuss and agree the way forward before changing the current implementations.